src/main/java/com/minecolonies/coremod/colony/workorders/WorkOrderBuildDecoration.java

```diff
@@ -37,17 +69,282 @@ public WorkOrderBuildDecoration(final String structureName, final String workOrd
+    {
+        boolean sendMessage = true;
+        boolean hasBuilder = false;
+
+        for (@NotNull final CitizenData citizen : colony.getCitizens().values())
+        {
+            final JobBuilder job = citizen.getJob(JobBuilder.class);
+
+            if (job == null || citizen.getWorkBuilding() == null)
+            {
+                continue;
+            }
+
+            hasBuilder = true;
+
+            // don't send a message if we have a valid worker that is busy.
+            if (canBuild(citizen))
+            {
+                sendMessage = false;
+            }
+
+            if (job.hasWorkOrder())
+            {
+                continue;
+            }
+
+            //  A Build WorkOrder may be fulfilled by a Builder as long as any ONE of the following is true:
+            //  - The Builder's Work AbstractBuilding is built
+            //  - OR the WorkOrder is for the Builder's Work AbstractBuilding
+            //  - OR the WorkOrder is for the TownHall
+            if (canBuild(citizen))
+            {
+                job.setWorkOrder(this);
+                this.setClaimedBy(citizen);
+                return;
+            }
+        }
+
+        sendBuilderMessage(colony, hasBuilder, sendMessage);
+    }
+
+    protected void sendBuilderMessage(@NotNull final Colony colony, final boolean hasBuilder, final boolean sendMessage)
+    {
+        if (hasSentMessageForThisWorkOrder)
+        {
+            return;
+        }
+
+        if (!hasBuilder)
+        {
+            hasSentMessageForThisWorkOrder = true;
+            LanguageHandler.sendPlayersMessage(colony.getMessageEntityPlayers(),
+              "entity.builder.messageNoBuilder");
+        }
+    }
+
+    /**
+     * Checks if a builder may accept this workOrder.
+     *
+     * @param citizen which could build it or not
+     * @return true if he is able to.
+     */
+    protected boolean canBuild(@NotNull final CitizenData citizen)
+    {
+        return true;
+    }
+
+
     @Override
     public boolean isValid(final Colony colony)
     {
         return true;
     }
 
+    @NotNull
+    @Override
+    protected WorkOrderType getType()
+    {
+        return WorkOrderType.BUILD;
+    }
+
+
     @Override
     protected String getValue()
     {
         return workOrderName;
     }
+
+    /**
+     * Returns the ID of the building (aka ChunkCoordinates).
+     *
+     * @return ID of the building.
+     */
+    public BlockPos getBuildingLocation()
+    {
+        return buildingLocation;
+    }
+
+    /**
+     * Get the name the structure for this work order.
+     *
+     * @return the internal string for this structure.
+     */
+    public String getStructureName()
+    {
+        return structureName;
+    }
+
+    /**
+     * Gets how many times this structure should be rotated.
+     *
+     * @param world where the decoration is
+     * @return building rotation.
+     */
+    public int getRotation(final World world)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "world". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1172)